fontforge/splinestroke.c

```diff
@@ -708,7 +708,7 @@ return;		/* Essentially colinear */ /* Won't be perfect because control points l
 	    p->needs_point_left = p->needs_point_right = false;
 	    p->left_hidden = bends_left;
 	    p->right_hidden = !bends_left;
-	    if ( rot.x<=fabs(diff_angle.x) ) {
+	    if ( rot.x<=diff_angle.x || (diff_angle.x <= -1 && rot.x <= -0.999999) ) { /* close enough */
```

Comment:
Looking at the code, it appears the ranges are [-1, 1] for `diff_angle.x` and [-1, 1] for `rot.x`. Now, `rot.x` will always [start in the range of [0, 1]](https://github.com/jtanx/fontforge/blob/e7ff15b015c72cef235fe4c55449c54f59125559/fontforge/splinestroke.c#L684). In the subsequent loops, `rot.x` is gradually decreased until it is less than `diff_angle.x`, which is the termination condition.

However, for what I presume is due to rounding errors, `diff_angle.x` can never reach -1. Instead, it will approach close to -1 before increasing again (it oscillates in the range of (-1,1)), hence causing the infinite loop.

In this case, I think an asymmetric fix (instead of say, `fabs(diff_angle.x-rot.x) < 0.0001`) is OK, since the opposite case of `diff_angle.x=1` won't cause the same issue.